frontend/src/scenes/surveys/SurveyResponseFilters.tsx

```diff
@@ -115,7 +115,7 @@ function _SurveyResponseFilters(): JSX.Element {
                             const operators = OPERATOR_OPTIONS[question.type] || []
 
                             return (
-                                <React.Fragment key={question.originalQuestionIndex}>
+                                <React.Fragment key={question.id ?? question.originalQuestionIndex}>
```

Comment:
`originalQuestionIndex` is removed from the SDK right [PostHog/posthog-js@`e181e41` (#1764)](https://github.com/PostHog/posthog-js/pull/1764/commits/e181e4129786d15c40bd5c413155b24882f47448)
so we should not use it anymore, or do we need it for back compatibility?

Reply:
is `originalQuestionIndex` here coming from the JS SDK or the types defined in posthog?

Reply:
[it's coming from here (unrelated to SDK)](https://github.com/PostHog/posthog/blob/8dc41e0f407399ce56767e06e09963045b813646/frontend/src/scenes/surveys/SurveyResponseFilters.tsx#L90-L96). those are unrelated, I just used the same name since they serve the same purpose

Reply:
so can we remove this now then? `originalQuestionIndex` should be removed from all FE/BE bits since its not used at all, it was just leaking from the SDK which was used only locally

Reply:
we can't, i still need it to construct the filters for responses created pre-question ID

Reply:
that can be inferred from the index position since the value cached in the database is useless

Reply:
or just rename it to avoid the confusion and clash with older versions of the SDK

Reply:
sure i'll rename it then

Reply:
> it can't, because first I'm [filtering to remove all questions that are not filterable](https://github.com/PostHog/posthog/blob/chore/use-question-id-instead-of-question-index-for-survey-queries/frontend/src/scenes/surveys/SurveyResponseFilters.tsx#L89-L100). what do you mean value cached in the DB? this is calculated on this file (FE), when iterating on the questions to build the filters

that was the problem around all the confusion with `originalQuestionIndex`, it was calculated in the SDK and leaking to the database (now removed).
now we are using the very same attribute name on the frontend, which brings back the confusion since the name has no semantic.
`questionIndex` would avoid the confusion and be semantically correct